### DISCUSSION TITLE
Warning when targeting cuda backend

### DISCUSSION THREAD

**Discussion by Luigi-Crisci** [intel/llvm#4610] [Q&A] [answered]
Hello,
When I try to build a sycl program targeting the cuda backend, I get this warning.
```
Warning: linking module '/home/lcrisci/sycl_workspace/llvm/build/lib/clang/14.0.0/../../clc/remangled-l64-signed_char.libspirv-nvptx64--nvidiacl.bc': Linking two modules of different target triples: '/home/lcrisci/sycl_workspace/llvm/build/lib/clang/14.0.0/../../clc/remangled-l64-signed_char.libspirv-nvptx64--nvidiacl.bc' is 'nvptx64-unknown-nvidiacl' whereas 'sycl_test.cpp' is 'nvptx64-nvidia-cuda'
```
The code works, but I'm not getting what is going on.

I builded the last version for the sycl branch following the Get Started Guide, using Cuda 10.2

**Comment by bader** [comment, marked as answer]
@Luigi-Crisci, this looks like a compiler issue.

@Naghasan, @AerialMantis, FYI.

A couple of thoughts:
1. I'm not sure if we can set 'nvptx64-nvidia-cuda' triple while building libclc, but we could update the triple in libclc-remangler tool. Right?
2. Another option is to suppress llvm-link warnings with a flag as we do for [libdevice](https://github.com/intel/llvm/blob/sycl/clang/lib/Driver/ToolChains/SYCL.cpp#L237-L239). I suppose it's not applied to for libclc.

**Reply by Luigi-Crisci** [reply]
Thanks @bader for the fast answer!

A thing that I want to clarify is that i want to target Nvidia backend directly: it should not invole openCL at all, am I right? Or am I missing something?

**Reply by bader** [reply]
It doesn't involve OpenCL, but math library available from the device code is written in OpenCL C language.

**Reply by Naghasan** [reply]
I've notice that as well. A fairly recent upstream change made linked builtin modules report issues. They were previously never reported, but this has always been present.

libclc and CUDA libdevice are impacted as they don't have the same triple. AFAIK, there is no direct equivalent to the suppress warning.

It is not clean, but this can be totally be ignored. The libspirv is designed around it being used for SYCL and libdevice is made for the NVPTX backend.

**Reply by keryell** [reply]
Adding an option to remove the warning or preprocess the `libclc` into a version with the right triple?
